cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1200,6 +1207,122 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 	return nil
 }
 
+func generateMountsModeFactoryReset(mst *initramfsMountsState) error {
+	// steps 1 and 2 are shared with install mode
+	model, snaps, err := generateMountsCommonInstallRecover(mst)
+	if err != nil {
+		return err
+	}
+	// get the disk that we mounted the ubuntu-seed partition from as a
+	// reference point for future mounts
+	disk, err := disks.DiskFromMountPoint(boot.InitramfsUbuntuSeedDir, nil)
+	if err != nil {
+		return err
+	}
+	// step 3: find ubuntu-save, unlock and mount, note that factory-reset
+	// mode only cares about ubuntu-save, as ubuntu-data and ubuntu-boot
+	// will be wiped anyway so we do not even bother looking up those
+	// partitions (which may be corrupted too, hence factory-reset was
+	// invoked)
+	degradedState := recoverDegradedState{}
+	part := degradedState.partition("ubuntu-save")
+	degraded := false
+	isEncrypted := false
+	_, findErr := disk.FindMatchingPartitionWithFsLabel(secboot.EncryptedPartitionName("ubuntu-save"))
+	if findErr == nil {
```

Comment:
can't we use/call recoverModeStateMachine.unlockEncryptedSaveFallbackKey|mountSave|openUnencryptedSave for some of this?

Reply:
Actually, I've refactored to use the recovery state machine, but starting from the lookup of ubuntu save in the fallback key path.